crates/sovereign-sdk/module-system/sov-modules-core/src/module/spec.rs

```diff
@@ -53,6 +53,9 @@ pub trait Spec: BorshDeserialize + BorshSerialize {
     /// Authenticated state storage used by the rollup. Typically some variant of a merkle-patricia trie.
     type Storage: Storage + Send + Sync;
 
+    /// The type of LedgerDB (empty for zk).
+    type LedgerDB: Send + Sync + Clone;
```

Comment:
It can be a problem because the type must implement `SharedLedgerOps` or something like this. Please review and say how to implement it better.

Reply:
I agree it's going to be a problem. Since we're passing the generic type, i.e:
https://github.com/chainwayxyz/citrea/pull/2439/files#diff-f2899e9dfce24f4daea4e4c5c0ffd80d25a4507512aab6216c2fda717545cb9bR21

AFAICS We're not going to be able to use methods from that apart from `clone`... i think instead of defining an associated type here, we just return LedgerDB itself.

Reply:
To come to think of it, yes it should be fine.